### PR TITLE
INS-2436: Move TypeGetPendingRequests message handling to flow-powered handler/procedure

### PR DESCRIPTION
**Pull request by bigbes**

### CODE REVIEW

File: ledger/light/handle/get_pending_requests.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+		msg:      parcel.Message().(*message.GetPendingRequests),
+		replyTo:  rep,
+		reqPulse: parcel.Pulse(),
+	}
+}
+
+func (s *GetPendingRequests) Present(ctx context.Context, f flow.Flow) error {
+	jet := proc.NewFetchJet(*s.msg.DefaultTarget().Record(), flow.Pulse(ctx), s.replyTo)
+	s.dep.FetchJet(jet)
+	if err := f.Procedure(ctx, jet, true); err != nil {
```
Comment (andreyromancev) [inline ledger/light/handle/get_pending_requests.go]: false

File: ledger/light/handle/get_pending_requests.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+		return err
+	}
+
+	getPendingRequests := proc.NewGetPendingRequests(jet.Result.Jet, s.replyTo, s.msg, s.reqPulse)
+	s.dep.GetPendingRequests(getPendingRequests)
+	return f.Procedure(ctx, getPendingRequests, true)
```
Comment (andreyromancev) [inline ledger/light/handle/get_pending_requests.go]: false

File: ledger/light/proc/get_pending_requests.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+		reqPulse: reqPulse,
+	}
+}
+
+func (p *GetPendingRequests) Proceed(ctx context.Context) error {
+	p.replyTo <- p.reply(ctx)
+	return nil
+}
+
+func (p *GetPendingRequests) reply(ctx context.Context) bus.Reply {
```
Comment (andreyromancev) [inline ledger/light/proc/get_pending_requests.go]: No need for a separate function.

File: ledger/light/proc/get_pending_requests.go [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+	msg := p.msg
+	jetID := insolar.ID(p.jet)
+
+	hasPendingRequests := false
+	pendingStorage := p.Dep.RecentStorageProvider.GetPendingStorage(ctx, jetID)
+	for _, reqID := range pendingStorage.GetRequestsForObject(*msg.Object.Record()) {
+		if reqID.Pulse() < p.reqPulse {
+			hasPendingRequests = true
+		}
```
Comment (andreyromancev) [inline ledger/light/proc/get_pending_requests.go]: `break`?

File: ledger/light/handle/get_pending_requests.go (hunk 1 above)
Comment (bigbes) [reply ledger/light/handle/get_pending_requests.go]: fixed

File: ledger/light/handle/get_pending_requests.go (hunk 2 above)
Comment (bigbes) [reply ledger/light/handle/get_pending_requests.go]: fixed

File: ledger/light/proc/get_pending_requests.go (hunk 3 above)
Comment (bigbes) [reply ledger/light/proc/get_pending_requests.go]: fixed

File: ledger/light/proc/get_pending_requests.go (hunk 4 above)
Comment (bigbes) [reply ledger/light/proc/get_pending_requests.go]: fixed

**Pull request merged**